src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -984,7 +985,7 @@ void processFrame(Http2Frame frame) throws IOException {
                     // always decode the headers as they may affect
                     // connection-level HPACK decoding state
                     if (orphanedConsumer == null || frame.getClass() != ContinuationFrame.class) {
-                        orphanedConsumer = new ValidatingHeadersConsumer();
+                        orphanedConsumer = new ValidatingHeadersConsumer(Context.RESPONSE);
```

Comment:
Should the context be `Context.REQUEST` if `hf instanceof PushPromiseFrame`?

Reply:
Good catch. Should be fixed now.